### DISCUSSION TITLE
How to parse m3u8 retrieved by videojs

### DISCUSSION THREAD

**Discussion by lastpeony** [videojs/video.js#8741] [Q&A]
Using
https://github.com/videojs/http-streaming
to play a hls stream.

How can i parse content of m3u8 file on client?
I saw
https://github.com/videojs/m3u8-parser
but there were only examples with local m3u8 files.

**Comment by logan858** [comment]
videojs can play m3u8 files already.

`videoJSOptions = {
    sources: [
        {
            src: "https://url/of/file.m3u8",
            type: "application/x-mpegURL"
        }
    ],
    etc, etc
}`

**Reply by lastpeony** [reply]
Hi
I know i need to parse the content i mean read m3u8 as string

**Comment by amtins** [comment]
@lastpeony what about something like:

```javascript
// I assume that the m3u8-parser library has been loaded.
const parser = new m3u8Parser.Parser();

fetch('https://d2zihajmogu5jn.cloudfront.net/bipbop-advanced/subtitles/eng/prog_index.m3u8')
  .then(r => r.text())
  .then(manifest => {
    parser.push(manifest);
    parser.end();

    const parsedManifest = parser.manifest;
    console.log(parsedManifest);
  }); 
```
